src/network/MissionControlProtocol.cpp

```diff
@@ -88,65 +97,57 @@ static bool validateDriveModeRequest(const json& j) {
 
 void MissionControlProtocol::handleDriveModeRequest(const json& j) {
 	std::string mode = j["mode"];
+	control::swerve_rots_t target;
 	if (mode == "normal") {
-		swerveController.driveMode.first = DriveMode::Normal;
-		for (int i = 0; i < 4; i++) {
-			robot::setMotorPos(Constants::Drive::WHEEL_IDS[i],
-							   Globals::swerveController.WHEEL_ROTS.at(DriveMode::Normal)[i]);
-		}
+		target = swerveController.setDriveMode(DriveMode::Normal);
 	} else if (mode == "turn-in-place") {
-		swerveController.driveMode.first = DriveMode::TurnInPlace;
-		for (int i = 0; i < 4; i++) {
-			robot::setMotorPos(
-				Constants::Drive::WHEEL_IDS[i],
-				Globals::swerveController.WHEEL_ROTS.at(DriveMode::TurnInPlace)[i]);
-		}
+		target = swerveController.setDriveMode(DriveMode::TurnInPlace);
 	} else if (mode == "crab") {
-		swerveController.driveMode.first = DriveMode::Crab;
-		for (int i = 0; i < 4; i++) {
-			robot::setMotorPos(Constants::Drive::WHEEL_IDS[i],
-							   Globals::swerveController.WHEEL_ROTS.at(DriveMode::Crab)[i]);
-		}
+		target = swerveController.setDriveMode(DriveMode::Crab);
+	}
+	std::array<int, 4> target_rots = {target.lfRot, target.rfRot, target.lbRot, target.rbRot};
```

Comment:
Can we log a warning and no-op if `target` is uninitialized? i.e. if the mode was the wrong value. In theory this shouldn't happen bc of the validator, but a buggy validator shouldn't be able to damage the hardware if some uninitialized value is passed as the pid target.